py/tests/algorithms/anomaly_likelihood_test.py

```diff
@@ -329,7 +330,10 @@ def testSerialization(self):
 
 class AnomalyLikelihoodAlgorithmTest(TestCaseBase):
   """Tests the low-level algorithm functions"""
-
+  #seed for function _generateSampleData()
+  #zero means really random and not deterministic
```

Comment:
ok, this was for c++ htm::Random, can you please review if the same behavior is for numpy.random?

Reply:
For numpy.random zero is ordinary seed, but if you pass None, then it takes 'really' random

>  If seed is None, then RandomState will try to read data from /dev/urandom (or the Windows analogue) if available or seed from the clock otherwise.
[https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.RandomState.html?highlight=randomstate](url)

Tested in console:
`rng = numpy.random.RandomState(None)
rng.randint(5)
Out[35]: 2

rng = numpy.random.RandomState(None) 
rng.randint(5)
Out[37]: 0

rng = numpy.random.RandomState(None)
rng.randint(5)
Out[39]: 4`

Reply:
So i am not sure if we should use numpy.random instead of htm::Random... maybe it would be simplier, what do you think?

Reply:
Using htm::Random for just getting random integers and nothing else probably makes no sense if i am thinking about it...

Reply:
yes, htm::Random has limited functionality

Reply:
...maybe we cannot reach deterministic outputs for numpy's random (that's why we have htm::Random anyway), so instead of exact values, the limits for acceptance could only be adjusted?

Reply:
> 
> 
> yes, htm::Random has limited functionality

To make it clear, i ment it in this way: htm:Random is probably not needed, if numpy.random works

Reply:
> 
> 
> ...maybe we cannot reach deterministic outputs for numpy's random (that's why we have htm::Random anyway), so instead of exact values, the limits for acceptance could only be adjusted?

If i pass seed to numpy.random.RandomState(None) then i get deterministic output each time. But i don't know if that applies for all OS. Thats what you mean?

Reply:
> If i pass seed to numpy.random.RandomState(seed) then i get deterministic output each time. But i don't know if that applies for all OS. Thats what you mean?

yes, quite some algos are deterministic, but platform/implementation dependant. That's why all the bother with custom Random. 
But my main question is: do we need exactly deterministic results in these tests (same numbers), or can we just have a statistical test (the likelihood, mean etc are less than xxx)?

Reply:
Ok understood. I think we can have just statistical test and don't bother with custom Random. So i will use numpy.random. Thanks!